src/hott/01-paths.rzk.md

```diff
@@ -952,3 +965,151 @@ The following is the same as above but with alternating arguments.
+<g fill='#5c5cd6' transform='matrix(1.6 0 0 1.6 109.0036 1.652862)'>
+<use x='-26.334534' y='-1.334215' xlink:href='#g3-114'/>
+</g>
+<path d='M-50.73824-36.07805C-17.55072-65.30461 19.67968-65.30461 52.8672-36.07805' stroke='#000' fill='none' stroke-width='3.725984' stroke-miterlimit='10'/>
+<path d='M-50.73824-36.07805C-17.55072-65.30461 19.67968-65.30461 52.8672-36.07805' stroke='#fff' fill='none' stroke-width='2.4508' stroke-miterlimit='10'/>
+<g transform='matrix(1.6 0 0 1.6 31.3442 -59.615364)'>
+<use x='-26.334534' y='-1.334215' xlink:href='#g0-114'/>
+<use x='-22.629624' y='-1.334215' xlink:href='#g0-101'/>
+<use x='-18.924714' y='-1.334215' xlink:href='#g0-102'/>
+<use x='-15.219804' y='-1.334215' xlink:href='#g0-108'/>
+</g>
+<path d='M12.441408 11.519534L51.78512-18.42973' stroke='#000' fill='none' stroke-width='3.725984' stroke-miterlimit='10'/>
+<path d='M12.441408 11.519534L51.78512-18.42973' stroke='#fff' fill='none' stroke-width='2.4508' stroke-miterlimit='10'/>
+<path d='M23.0352 3.78907H41.19136V-10.699218H23.0352Z' fill='#fff'/>
+<g transform='matrix(1.6 0 0 1.6 68.6021 2.492462)'>
+<use x='-26.334534' y='-1.334215' xlink:href='#g3-72'/>
+</g>
+<path d='M1.066402-53.69533V-36.07805' stroke='#000' fill='none' stroke-width='3.725984' stroke-miterlimit='10'/>
+<path d='M1.066402-53.69533V-36.07805' stroke='#fff' fill='none' stroke-width='2.4508' stroke-miterlimit='10'/>
+<g transform='matrix(1.6 0 0 1.6 46.95037 -39.727964)'>
+<use x='-26.334534' y='-1.334215' xlink:href='#g3-72'/>
+<use x='-19.276269' y='-4.340732' xlink:href='#g1-48'/>
+</g>
+</g>
+</svg>
+
+```rzk
+#def eq-top-cancel-commutative-square
+  (A : U)
+  (v w y z : A)
+  (p : v = w)
+  (q : w = v)
+  (s : w = y)
+  (r : v = z)
+  (t : y = z)
+  (H : (concat A w v z q r) = (concat A w y z s t))
+  (H' : (concat A v w v p q) = refl)
+  : r = (concat A v w z p (concat A w y z s t))
+  :=
+    (concat
+    (v = z)
+    ( r )
+    ( concat A v v z refl r)
+    ( concat A v w z p (concat A w y z s t) )
+    (rev
+      (v = z)
+      (concat  A v v z refl r )
+      ( r )
+      (left-unit-concat A v z r))
+    (concat
+     ( v = z)
+     ( concat A v v z refl r)
+     ( concat A v v z (concat A v w v p q) r )
+     ( concat A v w z p (concat A w y z s t ))
+     (rev
+      ( v = z )
+      (concat A v v z  (concat A v w v p q) r )
+      (concat A v v z refl r)
+      ( concat-eq-left
+        ( A)
+        ( v)
+        ( v)
+        ( z)
+        ( concat A v w v p q )
+        ( refl )
+        ( H')
+        ( r) ))
+     ( concat
+       ( v = z)
+       ( concat A v v z (concat A v w v p q) r)
+       ( concat A v w z p (concat A w v z q r)  )
+       ( concat A v w z p (concat A w y z s t))
+       ( associative-concat A v w v z p q r)
+       ( concat-eq-right
+        ( A )
+        ( v )
+        ( w)
+        ( z)
+        ( p)
+        ( concat A w v z q r)
+        ( concat A w y z s t)
+        ( H)))))
```

Comment:
```suggestion
    ( rev
      ( v = z)
      ( concat  A v v z refl r)
      ( r)
      ( left-unit-concat A v z r))
    ( concat
     ( v = z)
     ( concat A v v z refl r)
     ( concat A v v z (concat A v w v p q) r)
     ( concat A v w z p (concat A w y z s t))
     ( rev
       ( v = z)
       ( concat A v v z  (concat A v w v p q) r)
       ( concat A v v z refl r)
       ( concat-eq-left
         ( A)
         ( v)
         ( v)
         ( z)
         ( concat A v w v p q)
         ( refl)
         ( H')
         ( r)))
     ( concat
       ( v = z)
       ( concat A v v z (concat A v w v p q) r)
       ( concat A v w z p (concat A w v z q r))
       ( concat A v w z p (concat A w y z s t))
       ( associative-concat A v w v z p q r)
       ( concat-eq-right
         ( A)
         ( v)
         ( w)
         ( z)
         ( p)
         ( concat A w v z q r)
         ( concat A w y z s t)
         ( H)))))
```

If you're using VSCode, you can try the [indent-rainbow](https://marketplace.visualstudio.com/items?itemName=oderwat.indent-rainbow) plugin to help you spot misindented code.